pkg/tsdb/postgres/postgres.go

```diff
@@ -47,20 +47,47 @@ func newPostgresQueryEndpoint(datasource *models.DataSource) (tsdb.TsdbQueryEndp
 	timescaledb := datasource.JsonData.Get("timescaledb").MustBool(false)
 
 	endpoint, err := sqleng.NewSqlQueryEndpoint(&config, &queryResultTransformer, newPostgresMacroEngine(timescaledb), logger)
-	if err == nil {
-		logger.Debug("Successfully connected to Postgres")
-	} else {
+	if err != nil {
 		logger.Debug("Failed connecting to Postgres", "err", err)
+		return nil, err
 	}
+
+	logger.Debug("Successfully connected to Postgres")
 	return endpoint, err
 }
 
 func generateConnectionString(datasource *models.DataSource, logger log.Logger) (string, error) {
+	logger.Debug("Trying to generate Postgres connection string")
+
 	sslMode := strings.TrimSpace(strings.ToLower(datasource.JsonData.Get("sslmode").MustString("verify-full")))
 	isSSLDisabled := sslMode == "disable"
 
-	// Always pass SSL mode
-	sslOpts := fmt.Sprintf("sslmode=%s", url.QueryEscape(sslMode))
+	var host string
+	var port int
+	if strings.HasPrefix(datasource.Url, "/") {
```

Comment:
This is better. Just came to think about that we've probably missed to validate unix socket in add/update data source validation, [code](https://github.com/grafana/grafana/blob/615b000acd0ae301a159e24bdc6e32f870aaf974/pkg/api/datasources.go#L135).